tests/binaries/heap-multiple-heaps.c

```diff
@@ -0,0 +1,47 @@
+/**
+ * -*- mode: c -*-
+ * -*- coding: utf-8 -*-
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+
+/* Any allocation over 128KB is directed directly to mmap, which we don't want. */
+#define LESS_THAN_MMAP_THRESHOLD    (127 * 1024)
+/* Experimentally, a new heap is created at ~500 allocations. */
+#define NUM_ALLOCS                  1032
```

Comment:
Where does this number come from?

Reply:
It's something of an appendix from initial experiments. When I created two threads (and therefore two arenas), I noticed that the second arena was 0x8000000 bytes away from the first arena. 0x8000000 / 127KB = ~1032, so I used that as a starting point for the number of allocations to create. As it turns out, that was bigger than needed - it only took 515 allocations to trigger the creation of a new heap, so I just left it as a "twice as big as the number of allocations I expect" limit.

I can either rename it to make its purpose (and arbitrary nature) clear, or rewrite the loop as a while loop like this:

```c
void *current, *previous;
current = previous = NULL;
while (1) {
    current = malloc(LESS_THAN_MMAP_THRESHOLD);
    int chunk_distance = (previous == NULL) ? current - previous : EXPECTED_CHUNK_DISTANCE;
    if (chunk_distance < 0 || chunk_distance > EXPECTED_CHUNK_DISTANCE) {
        __builtin_trap();
        break;
    }
}
```

Reply:
name is fine, value is fine, maybe you can clarify why ~500 allocations makes you choose 1032. e.g.

```c
/* Experimentally, a new heap is created at ~500 allocations. We double for a safe margin */
```

Reply:
@Grazfather can this be resolved?